### PR TITLE
Function tree refactor

### PR DESCRIPTION
**Pull request by spflueger**
The FunctionTree class was removed completely since it had no
functionality whatsoever and just pointlessly wrapped around the
TreeNode.
The construction of a Tree is now more straight forward. Before unique
names of the nodes were used to built up the tree. Which was very
confusing.
Now there are no more names and you simply add TreeNode objects to other
TreeNodes.
Also a lot of unneeded methods were removed.

### CODE REVIEW

File: Physics/HelicityFormalism/WignerD.hpp [hunk 1]
```diff
@@ -5,10 +5,10 @@
 #ifndef COMPWA_PHYSICS_HELICITY_FORMALISM_WIGNERD_HPP_
 #define COMPWA_PHYSICS_HELICITY_FORMALISM_WIGNERD_HPP_
 
-#include "Core/FunctionTree/FunctionTree.hpp"
 #include "Core/FunctionTree/Functions.hpp"
+#include "Core/FunctionTree/TreeNode.hpp"
 
-#include "qft++/WignerD.h"
+#include "ThirdParty/qft++/include/qft++/WignerD.h"
```
Comment (redeboer) [inline Physics/HelicityFormalism/WignerD.hpp]: Why include like this?

File: Tools/test/IntegrationTest.cpp [hunk 2]
```diff
@@ -117,10 +117,10 @@ const std::string JpsiDecayKinematics = R"####(
 )####";
 
 const std::string JpsiDecayTree = R"####(
-<Intensity Class='CoherentIntensity' Name='coherent_0'>
```
Comment (redeboer) [inline Tools/test/IntegrationTest.cpp]: Is backward compatible, right?

File: Physics/Dynamics/Voigtian.hpp [hunk 3]
```diff
@@ -97,10 +97,10 @@ inline std::complex<double> dynamicalFunction(double mSq, double mR, double wR,
   return result;
 }
 
-std::shared_ptr<ComPWA::FunctionTree::FunctionTree>
+std::shared_ptr<ComPWA::FunctionTree::TreeNode>
 createFunctionTree(InputInfo Params,
                    const ComPWA::FunctionTree::ParameterList &DataSample,
-                   unsigned int pos, std::string suffix);
```
Comment (redeboer) [inline Physics/Dynamics/Voigtian.hpp]: What was this for originally? Those unique names?

File: Examples/DalitzFit/DalitzFitApp.cpp [hunk 4]
```diff
@@ -75,7 +75,7 @@ std::string amplitudeModel = R"####(
 		    </Amplitude>
       </Amplitude>
     </Amplitude>
-	  <Amplitude Class="CoefficientAmplitude" Name="myAmp">
+	  <Amplitude Class="CoefficientAmplitude" Component="myAmp">
```
Comment (redeboer) [inline Examples/DalitzFit/DalitzFitApp.cpp]: Component is a new tag?

File: Physics/Dynamics/NonResonant.hpp [hunk 5]
```diff
@@ -1,24 +0,0 @@
-// Copyright (c) 2015, 2017 The ComPWA Team.
```
Comment (redeboer) [inline Physics/Dynamics/NonResonant.hpp]: Why removed?

File: Physics/HelicityFormalism/WignerD.hpp (hunk 1 above)
Comment (spflueger) [reply Physics/HelicityFormalism/WignerD.hpp]: We have a slightly customized version of qft++ and its a quite strict part of ComPWA

**Pull request merged**